napari/_qt/qt_main_window.py

```diff
@@ -194,6 +194,12 @@ def current_viewer(cls):
         window = cls.current()
         return window._qt_viewer.viewer if window else None
 
+    @classmethod
+    def all_open_viewers(cls):
+        viewers = [window._qt_viewer.viewer for window in cls._instances]
+        if viewers != []:
+            return viewers
```

Comment:
Why select `Optional[List[Viewer]]` instead of `List[Viewer]`?

There should also be a docstring.

Reply:
+1, I don't understand the reasoning of returning `None` instead of an empty list

Reply:
1. I didn't include a docstring because none of the other methods in this class had one. I can add one here, but in general very little of the qt code has docstrings (and I probably don't have time to do [another massive effort](https://github.com/napari/napari/pull/1022/files) to fix that)

2. Weak references would be a better idea (ditto for the `current_viewer` function)